packages/stark-core/src/logging/services/logging.service.ts

```diff
@@ -0,0 +1,288 @@
+"use strict";
+
+import * as uuid from "uuid";
+
+import { Serialize } from "cerialize";
+import { validateSync } from "class-validator";
+
+import { Store } from "@ngrx/store";
+import { Observable } from "rxjs/Observable";
+import { Subject } from "rxjs/Subject";
+
+import { Inject, Injectable } from "@angular/core";
+
+import { StarkLoggingService, starkLoggingServiceName } from "./logging.service.intf";
+import { StarkApplicationConfig, starkAppConfigConstantName } from "../../configuration/entities/index";
+import { StarkBackend } from "../../http/entities/backend/index";
+import { StarkCoreApplicationState } from "../../common/store/starkCoreApplicationState";
+import { StarkHttpStatusCodes } from "../../http/enumerators/index";
+import { StarkHttpHeaders } from "../../http/constants/index";
+// import {StarkXSRFService, starkXSRFServiceName} from "../../xsrf";
+import { StarkValidationErrorsUtil } from "../../util/validation-errors.util";
+import { StarkLogging, StarkLoggingImpl, StarkLogMessage, StarkLogMessageImpl, StarkLogMessageType } from "../entities/index";
+import { LogMessage, FlushLogMessages } from "../actions/index";
+
+const _noop: Function = require("lodash/noop");
+
+/**
+ * @ngdoc service
+ * @name stark-core.service:StarkLoggingService
+ * @description Basic logging service implementation.
+ * Integrates logging with the Redux store
+ *
+ * @requires ngrx-store.Store
+ * @requires StarkApplicationConfig
+ * @requires StarkXSRFService
+ */
+@Injectable()
+export class StarkLoggingServiceImpl implements StarkLoggingService {
+	private backend: StarkBackend;
+	private logUrl: string;
+	private logPersistSize: number;
+	private isPersisting: boolean;
+	private retryCounter: number;
+	private consoleDebug: Function;
+	private consoleInfo: Function;
+	private consoleWarn: Function;
+	private consoleError: Function;
+	private consoleTrace: Function;
+	private starkLogging: StarkLogging;
+	/** @internal */
+	private _correlationId: string;
+	// FIXME: uncomment these lines once XSRF Service is implemented
+	public constructor(
+		private store: Store<StarkCoreApplicationState>,
+		@Inject(starkAppConfigConstantName)
+		private appConfig: StarkApplicationConfig /*,
+					   @Inject(starkXSRFServiceName) private xsrfService: StarkXSRFService*/
```

Comment:
The XSRF service can be mocked when defining the LoggingModule so no need to comment this and its usage neither.

By doing so, we can just comment the @Inject(xxx) part because we will provide it manually (instead of Angular) to the service